scripts/skills/actives/rf_inspire_skill.nut

```diff
@@ -0,0 +1,99 @@
+		this.m.IconDisabled = "skills/perk_28_active_sw.png";	// unused vanilla artwork
+		this.m.Overlay = "rf_inspire_skill";
+		this.m.SoundOnUse = [];		// Todo add sound
+		this.m.Type = ::Const.SkillType.Active;
+		this.m.Order = ::Const.SkillOrder.UtilityTargeted;
+		this.m.IsSerialized = false;
+		this.m.IsActive = true;
+		this.m.IsTargeted = true;
+		this.m.IsStacking = false;
+		this.m.IsIgnoredAsAOO = true;
+		this.m.ActionPointCost = 3;
+		this.m.FatigueCost = 15;
+		this.m.MinRange = 1;
+		this.m.MaxRange = 3;
+		// this.m.AIBehaviorID = ::Const.AI.Behavior.ID.RF_Inspire;	// Todo Implement
+	}
+
+	function getTooltip()
+	{
+		local ret = this.skill.getDefaultUtilityTooltip();
+
+		ret.extend([
+			{
+				id = 10,
+				type = "text",
+				icon = "ui/icons/bravery.png",
+				text = ::Reforged.Mod.Tooltips.parseString("Trigger a positive [Morale Check|Concept.Morale] with a bonus of " + ::MSU.Text.colorizeFraction(this.m.InspireBonusFraction) + " (" + ::MSU.Text.colorizeValue(this.getInspireBonus()) + ") of your [Resolve|Concept.Bravery] on the target"),
+			},
+			{
+				id = 15,
+				type = "text",
+				icon = "ui/icons/vision.png",
+				text = "Has a range of " + ::MSU.Text.colorGreen(this.getMaxRange()) + " tiles",
+			},
+		]);
+
+		return ret;
+	}
+
+	function onVerifyTarget( _originTile, _targetTile )
+	{
+		if (!this.skill.onVerifyTarget(_originTile, _targetTile)) return false;
+
+		local target = _targetTile.getEntity();
+		if (this.getContainer().getActor().getFaction() != target.getFaction()) return false;
+		return this.__canBeInspired(target);
+	}
+
+	function onUse( _user, _targetTile )
+	{
+		local target = _targetTile.getEntity();
+
+		target.checkMorale(1, this.getInspireBonus(), ::Const.MoraleCheckType.Default, "perk_rf_inspire");	// TODO make art
+		::Tactical.Entities.getFlags().set(target.getID() + this.m.InspireFlag, ::Time.getRound());	// We use entity flags to keep track of this skill's usage instead of a dummy skill
+
+		return true;
+	}
+
+// New Functions
+	function getInspireBonus()
+	{
+		local inspireBonus = this.getContainer().getActor().getCurrentProperties().getBravery() * this.m.InspireBonusFraction;
+		return ::Math.max(0, inspireBonus);	// The bonus can never be negative
+	}
+
+	function __canBeInspired( _targetEntity )
+	{
+		if (target.getCurrentProperties().IsStunned) return false;
+		if (target.getMoraleState() >= ::Const.MoraleState.Confident) return false;
+		if (target.getMoraleState() >= target.m.MaxMoraleState) return false;	// This might be confusing for the player because it's not mentioned in the tooltip but it's just logical
+
+		return (this.__getLastInspiredRound(_targetEntity) < ::Time.getRound());
+	}
+
+	function __getLastInspiredRound( _targetEntity )
+	{
+		if (::Tactical.Entities.getFlags().has(_targetEntity.getID() + this.m.InspireFlag))
+		{
+			return ::Tactical.Entities.getFlags().get(_targetEntity.getID() + this.m.InspireFlag);
+		}
+
+		return 0;
```

Comment:
The design is to be only inspired once per turn, not per round. And we should keep it like that i.e. character dependent skills should generally always be turn-based and not round-based because of the concept of turns and rounds in this game. E.g. I inspire an ally, then that ally starts his turn, does some actions, then ends his turn. This should now qualify that ally again for an inspire.

While I really like the idea of using flags instead of dummy skill to block the usage of inspire on the same target again I don't know if it can achieve the turn-based design, as we need to know when the target has ended their turn. So, we might need to resort to a dummy skill unfortunately.

Reply:
I prefer a once-per-round design. That is cleaner to follow by the player and probably also to implement. Since there are slight differences between those two concepts I realize I then must update the tooltips to say round

Reply:
I see the advantage in simplicity of the round-based design. But I'm averse to the idea of round-based skills because that isn't in line with the concept of what a turn represents and what a round represents. The rest of the skills in this game all care about turns. So I quite strongly feel that this should be a turn-based effect. 

> Not to mention that you can otherwise inspire the same guy twice in the same round if you play with the turn order correctly.

That is fine. With the round-based design you can do something similar i.e. if the other guy acted after you in the current round and then in the next round you inspire him again before he acts. It's just shifted in time.

Reply:
For the actual 0.5.0 build I'd want the Encourage ability to be turn-based and not round-based. But in favor of speed I'm willing to let the round-based design be merged in for now so that we can get an unstable build out tonight for testers.